spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java

```diff
@@ -471,3 +489,18 @@ public CompositeEnvironmentRepository compositeEnvironmentRepository(
 	}
 
 }
+
+@Configuration(proxyBeanMethods = false)
+@Profile("secret-manager")
```

Comment:
it's not pretty common to do this. can you explain the reasoning behind this?

Reply:
Can you please ellaborate a bit more? I just copied the way you configure those in the class. I maybe wrong, we can change whatever you feel it is not OK.

Reply:
I saw another implementations, don't worry about this comment :)